plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergConfig.java

```diff
@@ -430,6 +439,58 @@ public IcebergConfig setSplitManagerThreads(int splitManagerThreads)
         return this;
     }
 
+    public boolean isManifestCachingEnabled()
+    {
+        return manifestCachingEnabled;
+    }
+
+    @Config("iceberg.io.manifest.cache-enabled")
+    @ConfigDescription("Enable/disable the manifest caching feature")
+    public IcebergConfig setManifestCachingEnabled(boolean manifestCachingEnabled)
+    {
+        this.manifestCachingEnabled = manifestCachingEnabled;
+        return this;
+    }
+
+    public DataSize getMaxManifestCacheSize()
+    {
+        return maxManifestCacheSize;
+    }
+
+    @Config("iceberg.io.manifest.cache.max-total-size")
+    @ConfigDescription("Maximum total amount to cache in the manifest cache")
+    public IcebergConfig setMaxManifestCacheSize(DataSize maxManifestCacheSize)
+    {
+        this.maxManifestCacheSize = maxManifestCacheSize;
+        return this;
+    }
+
+    public Duration getManifestCacheExpireDuration()
+    {
+        return manifestCacheExpireDuration;
+    }
+
+    @Config("iceberg.io.manifest.cache.expiration-interval-duration")
+    @ConfigDescription("Maximum duration for which an entry stays in the manifest cache")
```

Comment:
Are manifest files mutable or this is mostly for releasing memory?

Reply:
immutable